token-client/README.md

```diff
@@ -35,15 +35,38 @@ The Resource owner password credentials (i.e., username and password) can be use
 Instantiate `XsuaaTokenFlows` with the `DefaultOAuth2TokenService` which
 makes use of [Apache HttpClient](https://hc.apache.org/):
 
+- Client secret based authentication method:
 ```java
 XsuaaTokenFlows tokenFlows = new XsuaaTokenFlows(
                                     new DefaultOAuth2TokenService(), 
-                                    new XsuaaDefaultEndpoints(<uaa_base_url>), 
+                                    new XsuaaDefaultEndpoints(<OAuth2ServiceConfiguration>), 
                                     new ClientCredentials(<client_id>, <client_secret>));
 ```
 The `DefaultOAuth2TokenService` can also be instantiated with a custom `CloseableHttpClient`.
 
-> The `<uaa_base_url>`, `<client_id>` and `<client_secret>` are placeholders for the information you get from the XSUAA service binding. 
+- X.509 based authentication method
+```java
+//Xsuaa managed certificate
+XsuaaTokenFlows tokenFlows = new XsuaaTokenFlows(
+                                    new DefaultOAuth2TokenService(HttpClient.create(<OAuth2ServiceConfiguration.getClientIdnetity()>)), 
+                                    new XsuaaDefaultEndpoints(<OAuth2ServiceConfiguration>), 
+                                    <OAuth2ServiceConfiguration.getClientIdentity()>);
+
+
+//Externally managed certificates
+ClientIdentity clientIdentity = new ClientCertificate(<certificate>,
```

Comment:
generally, instead of `<certificate>`/ `<client_id>` i would prefer `<OAuth2ServiceConfiguration>.getCertificate()` ...

BUT - how is it done in context of spring-xsuaa? 
There is no OAuth2ServiceConfiguration but XsuaaServiceConfiguration... This again would lead to a design change of XsuaaDefaultEndpoints which then would require another constructor with two parameters, base_url and cert_url. 

Lets discuss.